ambry-cloud/src/main/java/com/github/ambry/cloud/azure/StorageClientFactory.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2020  LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud.azure;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.ProxyOptions;
+import com.azure.core.http.netty.NettyAsyncHttpClientBuilder;
+import com.azure.core.util.Configuration;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.common.policy.RequestRetryOptions;
+import com.azure.storage.common.policy.RetryPolicyType;
+import com.github.ambry.config.CloudConfig;
+import com.microsoft.azure.cosmosdb.RetryOptions;
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.util.concurrent.ExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A factory class to create {@link BlobServiceClient} object.
+ */
+public abstract class StorageClientFactory {
+  Logger logger = LoggerFactory.getLogger(StorageClientFactory.class);
+
+  /**
+   * Create the {@link BlobServiceClient} object.
+   * @param {@link CloudConfig} object.
+   * @param {@link AzureCloudConfig} object.
+   * @return {@link BlobServiceClient} object.
+   * @throws MalformedURLException
+   * @throws InterruptedException
+   * @throws ExecutionException
+   */
+  public BlobServiceClient createBlobStorageClient(CloudConfig cloudConfig, AzureCloudConfig azureCloudConfig)
+      throws MalformedURLException, InterruptedException, ExecutionException {
```

Comment:
Rather than introducing several new checked exceptions that AzureBlobDataAccessor needs to know about, can we catch and rethrow as IllegalStateException similar to AzureCloudDestinationFactory.getCloudDestination()?

Reply:
done.